cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go

```diff
@@ -163,3 +210,10 @@ func runControlPlanePrepareKubeconfigPhaseLocal(c workflow.RunData) error {
 
 	return nil
 }
+
+// shouldDownloadCerts returns whether secrets should be downloaded for this join
+// process. This is true if we are joining a control plane and if a certificate key
+// was provided.
+func shouldDownloadCerts(data controlPlanePrepareData) bool {
+	return data.Cfg().ControlPlane != nil && len(data.CertificateKey()) > 0
```

Comment:
What about getting rid of this func (it is a simple boolean expression used only in one place...)